mesheryctl/pkg/utils/scripts.go

```diff
@@ -6,118 +6,203 @@ import (
 	"os/exec"
 )
 
-// TODO@Aisuko this function is so hard to read and maintain, please refactor it
-func GenerateConfigGKE(configPath, SAName, namespc string) error {
-	script := fmt.Sprintf(`
-	set -e
-	set -o pipefail
+// GKEConfig holds configuration for GKE script generation
 
-	KUBECFG_FILE_NAME="%s"
-	SERVICE_ACCOUNT_NAME=%s
-	NAMESPACE="%s"
-	TARGET_FOLDER=$(dirname ${KUBECFG_FILE_NAME})
+type GKEConfig struct {
+	ConfigPath string
+	SAName     string
+	Namespace  string
+}
 
-	command_exists() {
-		command -v $1 >/dev/null 2>&1
-	}
-	check_prerequisites() {
-		if ! command_exists jq ; then
-			echo -e "\\nMissing required utility: 'jq'. Please install 'jq' and try again."
-			exit;
-		fi
-		if ! command_exists base64 ; then
-			echo -e "\\nMissing required utility: 'base64'. Please install 'base64' and try again."
-			exit;
-		fi
-		if ! command_exists awk ; then
-			echo -e "\\nMissing required utility: 'awk'. Please install 'awk' and try again."
-			exit;
-		fi
-		if ! command_exists tail ; then
-			echo -e "\\nMissing required utility: 'tail'. Please install 'tail' and try again."
-			exit;
-		fi
+// GenerateConfigGKE generates kubernetes config for GKE
+func GenerateConfigGKE(configPath, SAName, namespc string) error {
```

Comment:
```suggestion
// GenerateConfigGKE generates kubernetes config for GKE
func GenerateGKEKubernetesConfig(configPath, SAName, namespace string) error {
```

Reply:
@lekaf974 , it seems like the function is only referred to as GenerateConfigGKE in other files. I tried renaming it, but I encountered these errors during the make process. See the screenshot below for details:
<img width="384" alt="Screenshot 2024-11-14 at 3 55 34 AM" src="https://github.com/user-attachments/assets/73522568-f0fa-46a8-b147-a17763d36dd9">